Lib/fontbakery/checks/base_has_width.py

```diff
@@ -5,14 +5,20 @@
 
 
 def is_space(codepoint):
-    return unicodedata.category(chr(codepoint)) in [
-        "Zs",  # Space Separator
-        "Zl",  # Line Separator
-        "Zp",  # Paragraph Separator
-        "Cf",  # Format
-        "Mn",  # Nonspacing Mark
-        "Cc",  # Control
-    ]
+    return (
+        unicodedata.category(chr(codepoint))
+        in [
+            "Zs",  # Space Separator
+            "Zl",  # Line Separator
+            "Zp",  # Paragraph Separator
+            "Cf",  # Format
+            "Mn",  # Nonspacing Mark
+            "Cc",  # Control
+        ]
+        or 0xE000 <= codepoint <= 0xF8FF
+        or 0xF0000 <= codepoint <= 0xFFFFD
+        or 0x100000 <= codepoint <= 0x10FFFD
```

Comment:
@simoncozens, I think this (ignoring PUA) would be a change to be ported to fontspector as well.